includes/class-llms-media-protector.php

```diff
@@ -0,0 +1,640 @@
+		if ( ! $file_size ) {
+			return;
+		}
+
+		$media_file   = get_post( $media_id );
+		$content_type = $media_file->post_mime_type;
+
+		header( "Content-Type: $content_type" );
+		header( "Content-Length: $file_size" );
+		header( 'X-Robots-Tag: noindex, nofollow', true );
+	}
+
+	/**
+	 * Sends a header that redirects the HTTP client to the media file's URL.
+	 *
+	 * This method calls the {@see https://www.php.net/manual/function.exit.php `exit()`} function and does not return.
+	 *
+	 * @since [version]
+	 *
+	 * @param int $media_id The post ID of the media file.
+	 * @return void
+	 */
+	protected function send_redirect( int $media_id ) {
+
+		$url = $this->get_media_url( $media_id );
+		header( "Location: $url" );
+		exit();
+	}
+
+	/**
+	 * Serves the requested media file to the HTTP client.
+	 *
+	 * This method calls the {@see https://www.php.net/manual/function.exit.php `exit()`} function and does not return.
+	 *
+	 * @todo Maybe respond with 404 Not Found?
+	 * @todo Maybe respond with 401 Unauthorized?
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function serve_file() {
+
+		$media_id = $_GET[ self::QUERY_PARAMETER_ID ];
+		$media_file = get_post( $media_id );
+
+
+		// Validate that the attachment post exists.
+		if ( is_null( $media_file ) ) {
+			return;
+		}
+
+		$file_name = $this->get_media_path( $media_id );
+
+		$size = null;
+		if ( isset( $_GET[ self::QUERY_PARAMETER_SIZE ] ) ) {
+			if ( '[' === $_GET[ self::QUERY_PARAMETER_SIZE ][0] ) {
+				$size = json_decode( $_GET[ self::QUERY_PARAMETER_SIZE ] );
+			} else {
+				$size = $_GET[ self::QUERY_PARAMETER_SIZE ];
+			}
+		}
+
+		$icon = null;
+		if ( isset( $_GET[ self::QUERY_PARAMETER_ICON ] ) ) {
+			$icon = (bool) $_GET[ self::QUERY_PARAMETER_ICON ];
+		}
+
+		// Optionally, use an alternate image size.
+		if ( ! is_null( $size ) || ! is_null( $icon ) ) {
+			$image     = wp_get_attachment_image_src( $media_id, $size, $icon );
+			$base_name = basename( $image[0] );
+			$file_name = dirname( $file_name ) . "/$base_name";
+		}
+
+		// Validate that the media file exists.
+		if ( false === file_exists( $file_name ) ) {
+			return;
+		}
+
+		// Is the user authorized to view the file?
+		$is_authorized = $this->is_authorized_to_view( get_current_user_id(), $media_id );
+		if ( false === $is_authorized ) {
+			$content_type = $media_file->post_mime_type;
+			if ( 0 === stripos( $content_type, 'image/' ) ) {
+				// @todo Find or create an image to denote unauthorized access to an image file.
+				$file_name = LLMS_PLUGIN_DIR . 'assets/images/circle-with-a-slash.png';
+			} else {
+				header( 'HTTP/1.1 403 Forbidden' );
+				exit();
+			}
+		}
+
+		/**
+		 * Determine how the media file should be served.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $serve_method One of the {@see LLMS_Media_Protector::SERVE_REDIRECT LLMS_Media_Protector::SERVE_X} constants.
```

Comment:
```suggestion
		 * @param string $serve_method One of the {@see LLMS_Media_Protector::SERVE_X} constants.
```

Reply:
I wanted the `@see` tag to link to the default serve method constant, which has more information. `SERVE_X` doesn't link. How about this?
```suggestion
		 * @param string $serve_method One of the LLMS_Media_Protector::SERVE_X constants, {@see LLMS_Media_Protector::SERVE_REDIRECT}.
```

Another option would be to use a `@see` link to each of the serve constants.

Reply:
I don't really care... I thought you had made a mistake with the initial version